lessons/projects/5.mdx

```diff
@@ -0,0 +1,1379 @@
+
+Doesn't it feel like we are still missing something?
+
+Right! We want our users to see information when they minting. We want 
+them to know when a mint *is in progress* and we want them to see the 
+result once the mint *is successful*.
+
+But what is this *modal*. It is a an element that is usually used as 
+an overlay that provides its own interaction environment separate from 
+the page below it.
+
+Okay lets write some code for it.
+
+To be able to display information about our latest mint, we must first
+get information about it from our smart contract.
+
+```jsx
+// imports, consts and WAGMI useContractWrite hook...
+
+  const {
+    data: tokenData,
+    refetch: refetchTokenData,
+  } = useContractRead({
+    addressOrName: CONTRACT_ADDRESS,
+    contractInterface: TierABI.abi,
+    functionName: "totalSupply",
+    watch: true,
+  })
+
+  const {
+    data: tokenURI,
+  } = useContractRead({
+    addressOrName: CONTRACT_ADDRESS,
+    contractInterface: TierABI.abi,
+    functionName: "tokenURI",
+    args: tokenData,
+    watch: true,
+  })
+
+  // useEffect to check user connection...
+
+  useEffect(() => {
+    try {
+      if (tokenURI) {
+        setLatestNFTMinted(JSON.parse(window.atob(tokenURI.substring(tokenURI.indexOf(',') + 1))));
+      }
+    } catch (error) {
+      console.log("Error fetching token URI", error.message);
+    }
+  }, [tokenData, tokenURI]);
+
+  const mintToken = async (e) => {
+    try {
+      // mint txn function
+
+      await mintTxn.wait();
+      console.log("This is the mint data", mintData);
+      refetchTokenData();
+      
+      // catch condition for mintToken...
+  };
+
+  // return statement...
+```
+
+- The first thing we need is the `tokenId` of the latest NFT minted. We 
+can do so by reading the value of the `totalSupply` stored in our smart 
+contract because the `totalSupply` should equal to the latest `tokenId` in
+our case. We can fetch this using the `useContractRead` hook from WAGMI 
+that is configured for reading data from smart contracts. The 
+`refetchTokenData` function will be called in our `mintToken` function 
+to update the `tokenData` after minting has succeeded. We pass a 
+`watch: true` key-value pair so that the hook keeps and active watch on 
+the value of totalSupply.
+- Then the returned value (tokenData) is passed in as an argument to the 
+`tokenURI` function from our smart contract and we received a `Base64` 
+encoded string in return.
+- A `useEffect` hook keeps updating this string everytime the `tokenData` 
+or `tokenURI` is updated and stores a decoded JSON format of it in the 
+`latestNFTMinted` variable.
+
+We're almost there. We just need to do two more checks before we actually
+look at our modal. The first is that we must display our modal only when 
+we want it to and the other is to display a loading message until our NFT 
+has successfully minted. 
+
+```jsx 
+// useEffect for tokenURI...
+
+  const mintToken = async (e) => {
```

Comment:
If you want me to use this code to replace something that has come before, please tell me?